models/auth/twofactor.go

```diff
@@ -58,11 +59,13 @@ func init() {
 
 // GenerateScratchToken recreates the scratch token the user is using.
 func (t *TwoFactor) GenerateScratchToken() (string, error) {
-	token, err := util.RandomString(8)
+	tokenBytes, err := util.SecureRandomBytes(6)
 	if err != nil {
 		return "", err
 	}
-	t.ScratchSalt, _ = util.RandomString(10)
+	const base32Chars = "ABCDEFGHJKLMNPQRSTUVWXYZ23456789"
+	token := base32.NewEncoding(base32Chars).WithPadding(base32.NoPadding).EncodeToString(tokenBytes)
```

Comment:
This poses a bigger question for me personally, base{32,64} or hex encoding? When I did https://github.com/go-gitea/gitea/pull/18179 I'd chose for Hex to do the encoding because of the constant size length(2 * src), IIRC Base32 is only 1.6 * src (or something in that area). I think we should go for a uniform path to do the encoding of bytes as mixing base and hex is a bit confusing for me as it doesn't seem to have a clear advantage or disadvantage to do only of of the 2.

Reply:
Dammit, missed the comments of ambiguous characters. In that case I think we can move this to a util function(in the case we need this at another case).

Reply:
`the security level doesn't change. 6 * log(256) > 8 * log(62)`